src/braket/aws/aws_device.py

```diff
@@ -622,3 +666,133 @@ def get_device_region(device_arn: str) -> str:
                 f"Device ARN is not a valid format: {device_arn}. For valid Braket ARNs, "
                 "see 'https://docs.aws.amazon.com/braket/latest/developerguide/braket-devices.html'"
             )
+
+    def refresh_gate_calibrations(self) -> GateCalibrations:
+        """
+        Refreshes the gate calibration data upon request.
+
+        If the device does not have calibration data, None is returned.
+
+        Returns:
+            GateCalibrations: the calibration data for the device.
+        """
+        if self.gate_calibrations_href is not None:
+            try:
+                with urllib.request.urlopen(self.gate_calibrations_href.split("?")[0]) as f:
+                    json_calibration_data = self._parse_calibration_json(
+                        json.loads(f.read().decode("utf-8"))
+                    )
+                    return GateCalibrations(json_calibration_data)
+            except urllib.error.URLError:
+                raise urllib.error.URLError(f"Unable to reach {self.gate_calibrations_href}")
+        else:
+            return None
+
+    def _get_waveforms(self, waveforms_json: str) -> Dict:
+        waveforms = dict()
+        for waveform in waveforms_json:
+            w = waveforms_json[waveform]
+            if "amplitudes" in w.keys():
```

Comment:
- Why are we not checking w["name"] here? 
- Is it possible for you to clean up this code by dispatching to the appropriate method instead of using if/else conditions?
- Should the AwsDevice class know about all the different waveforms and where to dispatch or should it be the waveform class?